src/openrct2-ui/windows/Ride.cpp

```diff
@@ -1599,22 +1626,18 @@ static void window_ride_init_viewport(rct_window* w)
     }
     else if (viewSelectionIndex >= ride->num_vehicles && viewSelectionIndex < (ride->num_vehicles + ride->num_stations))
     {
-        int32_t count = viewSelectionIndex - ride->num_vehicles;
-        for (const auto& station : ride->stations)
+        try
         {
-            if (!station.Start.isNull())
-            {
-                count--;
-            }
-            if (count == 0)
-            {
-                auto location = station.GetStart();
-                focus.coordinate.x = location.x;
-                focus.coordinate.y = location.y;
-                focus.coordinate.z = location.z;
-                focus.sprite.type |= VIEWPORT_FOCUS_TYPE_COORDINATE;
-                break;
-            }
+            StationIndex stationIndex = GetStationIndexFromViewSelection(*w).value();
+            auto location = ride->stations[stationIndex].GetStart();
+            focus.coordinate.x = location.x;
+            focus.coordinate.y = location.y;
+            focus.coordinate.z = location.z;
+            focus.sprite.type |= VIEWPORT_FOCUS_TYPE_COORDINATE;
+        }
+        catch (const std::bad_optional_access& e)
```

Comment:
Although this is one way of doing this exceptions are expensive. Instead you can just check the validity with something like:
```cpp

            auto stationIndex = GetStationIndexFromViewSelection(*w);
            if (stationIndex){
            auto location = ride->stations[*stationIndex].GetStart();
}
            ```